subsys/nrf_security/src/drivers/cracen/cracenpsa/src/common.c

```diff
@@ -334,6 +334,547 @@ psa_status_t cracen_ecc_reduce_p256(const uint8_t *input, size_t input_size, uin
 }
 
+psa_status_t cracen_ecc_is_quadratic_residue(const sx_const_op *curve_prime,
+					     const sx_const_op *value,
+					     bool *is_qr)
+{
+#if PSA_VENDOR_ECC_MAX_CURVE_BITS > 0
+	int sx_status;
+	const size_t op_size = curve_prime->sz;
+
+	uint8_t tmp_buf1[PSA_BITS_TO_BYTES(PSA_VENDOR_ECC_MAX_CURVE_BITS)] = {};
+	uint8_t tmp_buf2[PSA_BITS_TO_BYTES(PSA_VENDOR_ECC_MAX_CURVE_BITS)] = {};
+
+	/**
+	 * value is a quadratic residue mod p if
+	 * value^((p-1)/2) mod p equals zero or one
+	 */
+
+	/* p_m_1 = (p - 1) */
+	cracen_be_add(tmp_buf1, op_size, 1); /* 1 */
+	if (cracen_be_sub(curve_prime->bytes, tmp_buf1, tmp_buf2, op_size) != 0) {
+		return PSA_ERROR_INVALID_ARGUMENT;
+	}
+	sx_op p_m_1 = {.sz = op_size, .bytes = tmp_buf2};
+
+	/* (p - 1) / 2 */
+	cracen_be_rshift(p_m_1.bytes, 1, tmp_buf1, op_size);
+	sx_const_op pm1_div_2 = {.sz = op_size, .bytes = tmp_buf1};
+
+	/* tmp = value^((p-1)/2) mod p */
+	sx_op tmp = {.sz = op_size, .bytes = tmp_buf2};
+
+	sx_status = sx_mod_exp(NULL, value, &pm1_div_2, curve_prime, &tmp);
+	if (sx_status != SX_OK) {
+		return silex_statuscodes_to_psa(sx_status);
+	}
+
+	safe_memzero(tmp_buf1, op_size);
+	cracen_be_add(tmp_buf1, op_size, 1); /* 1 */
+
+	*is_qr = constant_memcmp_is_zero(tmp.bytes, tmp.sz);
+	*is_qr = *is_qr || (constant_memcmp(tmp.bytes, tmp_buf1, tmp.sz) == 0);
+
+	return PSA_SUCCESS;
+#else
+	return PSA_ERROR_NOT_SUPPORTED;
+#endif /* PSA_VENDOR_ECC_MAX_CURVE_BITS > 0 */
+}
+
+#if PSA_VENDOR_ECC_MAX_CURVE_BITS > 0
+static psa_status_t cracen_get_sswu_z_brainpool_curve(size_t curve_bits, int *z)
+{
+	/**
+	 *  Unique curve parameter Z is taken from table 12.2 of
+	 *  the IEEE802.11 12.4.4.2.3
+	 */
+	switch (curve_bits) {
+	case 256:
+		/* IANA value (group): 28 */
+		*z = -2;
+		break;
+	case 384:
+		/* IANA value (group): 29 */
+		*z = -5;
+		break;
+	case 512:
+		/* IANA value (group): 30 */
+		*z = 7;
+		break;
+	default:
+		return PSA_ERROR_INVALID_ARGUMENT;
+	}
+
+	return PSA_SUCCESS;
+}
+
+static psa_status_t cracen_get_sswu_z_r1_curve(size_t curve_bits, int *z)
+{
+	/**
+	 *  Unique curve parameter Z is taken from table 12.2 of
+	 *  the IEEE802.11 12.4.4.2.3
+	 */
+	switch (curve_bits) {
+	case 192:
+		/* IANA value (group): 25 */
+		*z = -5;
+		break;
+	case 224:
+		/* IANA value (group): 26 */
+		*z = 31;
+		break;
+	case 256:
+		/* IANA value (group): 19 */
+		*z = -10;
+		break;
+	case 384:
+		/* IANA value (group): 20 */
+		*z = -12;
+		break;
+	case 521:
+		/* IANA value (group): 21 */
+		*z = -4;
+		break;
+	default:
+		return PSA_ERROR_INVALID_ARGUMENT;
+	}
+
+	return PSA_SUCCESS;
+}
+
+static psa_status_t cracen_get_sswu_z(psa_ecc_family_t curve_family,
+				      size_t curve_bits,
+				      int *z)
+{
+	switch (curve_family) {
+	case PSA_ECC_FAMILY_BRAINPOOL_P_R1:
+		return cracen_get_sswu_z_brainpool_curve(curve_bits, z);
+	case PSA_ECC_FAMILY_SECP_R1:
+		return cracen_get_sswu_z_r1_curve(curve_bits, z);
+	case PSA_ECC_FAMILY_MONTGOMERY:
```

Comment:
This is kind of a nit, but why you don't remove all these families and just  return PSA_ERROR_NOT_SUPPORTED as the default case?